postgresqltuner.pl

```diff
@@ -547,71 +555,75 @@ sub usage {
 		} else {
-			print_report_ok("Max possible memory usage for PostgreSQL is good");
+			print_report_ok("The potential max memory usage for PostgreSQL is adequate if the host is dedicated to it");
 		}
 		# track activity ratio
 		my $track_activity_ratio=$track_activity_size*100/$os->{mem_total};
 		if ($track_activity_ratio > 1) {
 			print_report_warn("Track activity reserved size is more than 1% of your RAM");
-			add_advice("track_activity","low","Your track activity reserved size is too high. Reduce track_activity_query_size and/or max_connections");
+			add_advice("track_activity","low","Your track activity reserved size is too high.  Reduce track_activity_query_size and/or max_connections");
 		}
 		# total ram usage with effective_cache_size
-		my $percent_mem_usage=($max_memory+$effective_cache_size)*100/$os->{mem_total};
-		print_report_info("max memory+effective_cache_size is ".format_percent($percent_mem_usage)." of total RAM");
+		my $percent_mem_usage=($max_memory+$effective_cache_size-$shared_buffers)*100/$os->{mem_total};
+		print_report_info("max memory + effective_cache_size - shared_buffers is ".format_percent($percent_mem_usage)." of the amount of RAM");
 		if ($percent_mem_usage < 60 and $shared_buffers_usage > 1) {
-			print_report_warn("Increase shared_buffers and/or effective_cache_size to use more memory");
+			print_report_warn("Increase shared_buffers to let PostgreSQL directly use more memory, especially if the machine is dedicated to PostgreSQL");
 		} elsif ($percent_mem_usage > 90) {
-			print_report_warn("the sum of max_memory and effective_cache_size is too high, the planner can find bad plans if system cache is smaller than expected");
+			print_report_warn("The sum of max_memory and effective_cache_size is too high, the planner may create bad plans because the system buffercache will probably be smaller than expected, especially if the machine is NOT dedicated to PostgreSQL");
 		}
 	}
 	# Hugepages
-	print_header_2("Huge pages");
-	if ($os->{name} eq 'darwin') {
-		print_report_unknown("No information on huge pages on MacOS.");
+	print_header_2("Huge Pages");
+  if (($os->{name} ne 'linux') && ($os->{name} ne 'freebsd')) { # not sure about FreeBSD
+		print_report_unknown("No Huge Pages on this OS.");
 	} else {
 		my $nr_hugepages=get_sysctl('vm.nr_hugepages');
 		if (!defined $nr_hugepages || $nr_hugepages == 0) {
 			print_report_bad("No Huge Pages available on the system");
 			last;
 		}
 		if (get_setting('huge_pages') eq 'on') {
-			print_report_ok("huge_pages enabled in PostgreSQL");
-		} else {
-			print_report_bad("huge_pages disabled in PostgreSQL");
-			add_advice("hugepages","medium","Enable huge_pages in PostgreSQL to consume system Huge Pages");
+			print_report_ok("huge_pages=on, therefore PostgreSQL needs Huge Pages");
+		}
+    elsif	(get_setting('huge_pages') eq 'try') {
+			print_report_bad("huge_pages=on, therefore PostgreSQL will try to use Huge Pages, if they are enabled");
```

Comment:
You have `try` set as a `print_report_bad` level message, while completely disabled is only `medium` advice.

If anything `on` should be a `print_report_warn`, since the server will completely fail to start if it has no available huge pages.

Reply:
You are right, thank you!